src/main/java/io/camunda/testing/assertions/ProcessInstanceAssertions.java

```diff
@@ -104,39 +97,36 @@ public ProcessInstanceAssertions isCompleted() {
    * @return this {@link ProcessInstanceAssertions}
    */
   public ProcessInstanceAssertions isNotCompleted() {
-    final Optional<Record<ProcessInstanceRecordValue>> processInstanceRecord =
-        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
-            .withProcessInstanceKey(actual.getProcessInstanceKey())
-            .withBpmnElementType(BpmnElementType.PROCESS)
-            .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED)
-            .stream()
-            .findFirst();
-
-    assertThat(processInstanceRecord.isPresent())
+    assertThat(isCompleted(actual.getProcessInstanceKey()))
         .withFailMessage("Process with key %s was completed", actual.getProcessInstanceKey())
         .isFalse();
-
     return this;
   }
 
+  /**
+   * Checks if a process instance has been completed
+   *
+   * @param processInstanceKey the key of the process instance
+   * @return boolean indicating whether the process instance has been completed
+   */
+  private boolean isCompleted(final long processInstanceKey) {
```

Comment:
:question: Is there ever a reason not to check for `actual.getProcessInstanceKey()`. In other words, why the paramter?

Reply:
I added the parameter so I could use the same name. I've changed the name to `isProcessInstanceCompleted` now.